### PR TITLE
config, kv: make TxnEntryCountLimit and TxnTotalSizeLimit configurable

### PR DESCRIPTION
**Pull request by jackysp**
### What problem does this PR solve? 
If TxnEntryCountLimit and TxnTotalSizeLimit are configurable, we could make the transaction commit more flexible.

### What is changed and how it works?
make TxnEntryCountLimit and TxnTotalSizeLimit configurable

### Check List 

Tests 

 - Unit test
 - Manual test (add detailed scripts or steps below)
1. Start a TiDB with txn-entry-count-limit = 100 / txn-total-size-limit = 102400
1.
```
create table t (i int); 
insert into t values (1),(2),(3);
insert into t select * from t; -- keep on doing it.
```

Code changes

 - Has exported variable/fields change

Side effects

 - Increased code complexity

### CODE REVIEW

**Comment by jackysp** [comment]
/run-all-tests tidb-test=release-2.0 tikv=release-2.0 pd=release-2.0

**Comment by tiancaiamao** [comment]
LGTM

File: config/config.toml.example [hunk 1]
```diff
@@ -143,6 +143,14 @@ query-feedback-limit = 1024
 # row count in statistics of a table is greater than it.
 pseudo-estimate-ratio = 0.7
 
+# The limitation of the number for the entries in one transaction.
+# If using TiKV as the storage, the entry represents a key/value pair.
```
Comment (winkyao) [inline config/config.toml.example]: Add a warning here，notice the user to change the item carefully.

File: config/config.toml.example [hunk 2]
```diff
@@ -143,6 +143,14 @@ query-feedback-limit = 1024
...
+txn-entry-count-limit = 300000
+
+# The limitation of the size in byte for the entries in one transaction.
+# If using TiKV as the storage, the entry represents a key/value pair.
+txn-total-size-limit = 104857600
```
Comment (winkyao) [inline config/config.toml.example]: ditto

**Pull request merged**